From 37741a7009bda3416917cdb1f2857f50e4a7812b Mon Sep 17 00:00:00 2001
From: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
Date: Fri, 14 Jun 2024 13:08:41 +0530
Subject: [PATCH] Move wifi NAN feature to mixins options

NAN and hotspot cannot work simultaneously, as both wants the
wifi chip tp be STA and AP simultaneously.

Move NAN to mixins options, whenever needed we can enable them in
mixins.spec and validate.

Tests done:
1. Build nan=false
2. Flash BM, check wifi and hotspot on success
3. Build nan=true
4. Flash BM, #>pm list features | grep wifi
5. Check wifi aware is listing in features list
6. Verify wifi aware service is running from logcat

Tracked-On: OAM-120724
Signed-off-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
---
 groups/wlan/iwlwifi/BoardConfig.mk | 3 +++
 groups/wlan/iwlwifi/option.spec    | 1 +
 groups/wlan/iwlwifi/product.mk     | 6 +++++-
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/groups/wlan/iwlwifi/BoardConfig.mk b/groups/wlan/iwlwifi/BoardConfig.mk
index 74bb579..8fc0ab0 100644
--- a/groups/wlan/iwlwifi/BoardConfig.mk
+++ b/groups/wlan/iwlwifi/BoardConfig.mk
@@ -12,4 +12,7 @@ BOARD_WPA_SUPPLICANT_PRIVATE_LIB ?= lib_driver_cmd_intc
 BOARD_SEPOLICY_DIRS += $(INTEL_PATH_SEPOLICY)/wlan/load_iwl_modules
 BOARD_SEPOLICY_DIRS += $(INTEL_PATH_SEPOLICY)/wlan/iwlwifi
 WIFI_HIDL_UNIFIED_SUPPLICANT_SERVICE_RC_ENTRY := true
+
+{{#nan}}
 WIFI_HIDL_FEATURE_AWARE := true
+{{/nan}}
diff --git a/groups/wlan/iwlwifi/option.spec b/groups/wlan/iwlwifi/option.spec
index b37b6fb..733fba0 100644
--- a/groups/wlan/iwlwifi/option.spec
+++ b/groups/wlan/iwlwifi/option.spec
@@ -10,3 +10,4 @@ libwifi-hal = false
 firmware =
 nvm = true
 iwl_upstream_drv = true
+nan = false
diff --git a/groups/wlan/iwlwifi/product.mk b/groups/wlan/iwlwifi/product.mk
index 3844942..e827b29 100644
--- a/groups/wlan/iwlwifi/product.mk
+++ b/groups/wlan/iwlwifi/product.mk
@@ -20,8 +20,12 @@ PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.hardware.wifi.xml:vendor/etc/permissions/android.hardware.wifi.xml \
     frameworks/native/data/etc/android.hardware.wifi.direct.xml:vendor/etc/permissions/android.hardware.wifi.direct.xml \
     frameworks/native/data/etc/android.software.ipsec_tunnels.xml:vendor/etc/permissions/android.software.ipsec_tunnels.xml \
-    frameworks/native/data/etc/android.hardware.wifi.rtt.xml:vendor/etc/permissions/android.hardware.wifi.rtt.xml \
+    frameworks/native/data/etc/android.hardware.wifi.rtt.xml:vendor/etc/permissions/android.hardware.wifi.rtt.xml
+
+{{#nan}}
+PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.hardware.wifi.aware.xml:vendor/etc/permissions/android.hardware.wifi.aware.xml
+{{/nan}}
 
 PRODUCT_PACKAGE_OVERLAYS += $(INTEL_PATH_COMMON)/wlan/overlay-disable_keepalive_offload
 
-- 
2.17.1

